Packs/FeedNVDv2/Integrations/FeedNVDv2/FeedNVDv2.py

```diff
@@ -0,0 +1,421 @@
+
+    relationships.extend([EntityRelationship(name="targets",
+                                             entity_a=cve_id,
+                                             entity_a_type="cve",
+                                             entity_b=product,
+                                             entity_b_type="software")
+                          .to_indicator() for product in products])
+
+    return list(vendors | products | parts), relationships
+
+
+def test_module(client: Client):
+    """
+    Performs a simple API call to NVD for CVE-2021-44228 using the provided API key
+
+    Args:
+        client: An instance of the BaseClient connection class
+        params: A dictionary containing HTTP parameters
+
+    Returns:
+        'ok' if a successful HTTP 200 message is returned
+
+    """
+    try:
+        client.http_get("/rest/json/cves/2.0/?noRejected", {})
+        return_results('ok')
+
+    except Exception as e:  # pylint: disable=broad-except
+        return_error("Invalid API key specified in integration instance configuration"
+                     + "\nError Message: " + str(e))
+
+
+def retrieve_cves(client, start_date: Any, end_date: Any, test_run: bool):
+    """
+    Iteratively retrieves CVEs from NVD from the specified modification date
+    through the date the fetch-indicators or nvd-get-indicators command is 
+    called
+
+    Args:
+        client: An instance of the BaseClient connection class
+
+    Returns:
+        Total number of CVE indicators fetched
+
+    """
+    url = "/rest/json/cves/2.0/?noRejected"
+    param = {}
+    start_index = 0
+    results_per_page = 2000
+    raw_cves = []  # type: ignore
+
+    start_date = datetime.fromisoformat(str(start_date))
+    end_date = datetime.fromisoformat(str(end_date))
+
+    param['lastModStartDate'] = start_date.strftime(DATE_FORMAT)  # type: ignore
+    param['lastModEndDate'] = end_date.strftime(DATE_FORMAT)  # type: ignore
+
+    if client.has_kev:
+        url += '&hasKev'
+
+    param['startIndex'] = int(start_index)  # type: ignore
+    param['resultsPerPage'] = results_per_page  # type: ignore
+
+    more_to_process = True
+
+    # Collect all the indicators together
+    while more_to_process:
+        try:
+            if not test_run:
+                res = client.http_get(url, param)
+            else:
+                with open('./Packs/FeedNVDv2/Integrations/FeedNVDv2/test_data/test_cve_data.json', encoding='utf-8') as f:
+                    res = json.loads(f.read())
+                    more_to_process = False
+            if "error" in res:
+                return_error(res.get('error'))
+            total_results = res.get('totalResults', 0)
+
+            if total_results:
+                raw_cves += res.get('vulnerabilities')
+
+                param['startIndex'] = str(int(param['startIndex']) + results_per_page)
+
+            if (int(param['startIndex']) >= total_results):
+                more_to_process = False
```

Comment:
@TerminalFin - You can just have:
```
while param['startIndex']<=total_results:
    ...
```

Even if startIndex sent is bigger than the result all you get is an emty list, so you can even just call the API until an empty list returns.